pkg/kv/kvserver/kvserverbase/bulk_adder.go

```diff
@@ -68,6 +68,11 @@ type BulkAdderOptions struct {
 	// the first buffer to pick split points in the hope it is a representative
 	// sample of the overall input.
 	InitialSplitsIfUnordered int
+
+	// ImportJobID specifies the import job the Bulk Adder is doing work for. If
+	// specified, the Bulk Adder's SSTBatcher will write the job ID to each
+	// versioned value's metadata.
+	ImportJobID int64
```

Comment:
Hm, an interesting wrinkle here is that we can restore into a different cluster, and `GenerateUniqueInt` isn't guaranteed to be unique across clusters, since it's based on the node ID and timestamp. That actually makes collisions far more likely than if they were completely random, since the values are very clustered.

This is mostly academic though, since it would need a pretty unlikely sequence of events.

Also, these IDs are going to be fairly large since they populate the higher-order bits of the ID (which means varints won't help). So we're adding on something like 10 bytes to each key.

Reply:
I'm not sure I follow. Per this [comment](https://github.com/cockroachdb/cockroach/pull/85138#discussion_r934733926), I think the restoring cluster should create new unique job IDs for each in progress import. I'll verify this plan in my comment with david.

Reply:
The "unique" ID is simply a concatenation of the node ID and current time, so it's only unique within a single cluster (because we guarantee clock monotonicity via the HLC). It's therefore possible (but unlikely) for e.g. two restores from two different imports in two different clusters to have the same ID, or for the target cluster to be running with a slow clock and generate the same ID.

I think this is mostly academic though, for a variety of reasons -- I mostly found it interesting/amusing that a random UUID would be far less likely to collide, because it isn't clustered around the current time and very-low-integer node IDs.

We don't need to do anything here.